kernel/src/scan/file_stream.rs

```diff
@@ -80,18 +80,15 @@ impl LogReplayScanner {
             None => actions,
         };
 
-        let schema_to_use = StructType::new(if is_log_batch {
-            vec![
-                crate::actions::schemas::ADD_FIELD.clone(),
-                crate::actions::schemas::REMOVE_FIELD.clone(),
-            ]
+        let schema_to_use = if is_log_batch {
+            get_log_schema().project_as_schema(&[ADD_NAME, REMOVE_NAME])?
```

Comment:
Because `project_as_schema` preserves schema order, this will only work because add comes before remove in the log schema. Could be a surprising pitfall worth a comment?